projects/Buttons_and_ADC/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+static void prv_button_interrupt_handler(const GpioAddress *adc_address, void *context) {
+  GpioAddress *adc_position = context;
+  uint16_t adc = 0;
+  adc_read_converted_pin(*adc_position, &adc);
```

Comment:
This is also a bit nitpicky but for production code, we always want to check what the output of the functions we call so that we know the data we are receiving is correct before outputting the data. If its not correct, we should LOG_DEBUG an error message
Eg
`if (adc_read_converted_pin(*adc_position, &adc) == STATUS_CODE_OK) {`